### ISSUE TITLE
[BUG] Two issues with Pac-Man Fever which may be linked

### ISSUE THREAD

**Issue by TheTechnician27** [PCSX2/pcsx2#4653]
As these issues both have to do with moving geometry, I figured putting them under one bug report was fine and may be more useful than separately.

**Describe the bug**
Bug 1) On the Tropical board, characters move across squares in a very jittery manner. Footage of this can be found [here.](https://www.youtube.com/watch?v=M1aW9bNpW-0&t=1m18s)

Bug 2) In the mini-game Gauntlet Golf, the drawbridges lower once and then close indefinitely. In the actual game, they open and close repeatedly, and they act as an obstacle for the golf ball. This renders the mini-game essentially unplayable, which is a big deal, as progression on boards is tied to performance in mini-games. Footage of what this looks like in the emulator can be found [here,](https://www.youtube.com/watch?v=-Zb5IroUR9M&t=6m14s), while footage of what this is supposed to look like can be found [here.](https://www.youtube.com/watch?v=67za8XnXK-0) The dragons and knights, however, still function normally in PCSX2.

These were recorded on a dev version of PCSX2 that's a few behind the current one, but testing it with the latest build (1592) returns the same results. I've tested both of these with and without MTVU just to be sure, and they were tested at default settings (i.e. native resolution) otherwise.

**To Reproduce**
Bug 1) Just advance on the Tropical board. The spaces that cause this problem seem to be deterministic, i.e. they're the same every time you play the board.

Bug 2) Just play Gauntlet Golf, either in the 'Mini-games' menu or on the Medieval board.

**Expected behavior**
Bug 1) The players are supposed to walk straight across the squares.

Bug 2) The drawbridge is supposed to open and close on a timed loop.

**GS Settings**
Default. D3D11 hardware. However, this seems more to do with the actual movement of the geometry itself and not how it's being rendered. I'll test this on software rendering when I have the time later just to make sure it's not a rendering issue.

**Emulation Settings**
All default settings. Tried with and without MTVU and with and without Instant VU1.

**System Info (please complete the following information):**
 -   PCSX2 Revision: dev-1592
 -   OS: Windows 10
 -   CPU: Ryzen 1700 @3.4 GHz
 -   GPU: GTX 1070

**Logs and Dumps**
I'm going to attach a GS dump here a bit later today when I have the time just in case this is something related to the rendering. For now, I thought it pertinent just to put the issue up.

**Comment by refractionpcsx2** [comment]
a GS Dump won't be an use to us, but a block dump will be

**Comment by refractionpcsx2** [comment]
A memory card save would be SUPER handy, nobody seems to have one online and the events are locked by default, I'm not playing through the entire game to unlock them :P

**Comment by TheTechnician27** [comment]
@refractionpcsx2 Sure thing. I can have the memory card save and the block dump uploaded in two-and-a-half hours, so I'll @ you when they're up. Do you want separate block dumps for the two different bugs, or is one okay?

**Comment by refractionpcsx2** [comment]
one is fine, just go in to the first event, then come out and go in to the second. The most important thing for me is that memcard

**Comment by TheTechnician27** [comment]
https://send.tresorit.com/a#hgCXM5n0wcmBUzrOa7TUDw

@refractionpcsx2 This contains the memcard and the block dump (I accidentally started the blockdump from boot; I turned it off in the debug menu, but it continued to write into the same file when I turned it back on, so it's all in one file). The password is just your username. Hopefully this helps. Probably something to do with the way the PS2 handles floats, amirite?

**Comment by refractionpcsx2** [comment]
Thanks.

Hopefully it's something fixable.

**Comment by refractionpcsx2** [comment]
OK for bug 2, set the VU rounding mode to Positive.  I'm struggling to reproduce bug 1, maybe it'll help that too?

**Comment by TheTechnician27** [comment]
I'll give that a try, and if it works, I can put it on the Wiki. Incidentally, the GameCube version of the game outright soft-locks (black screen) when you enter a mini-game on Dolphin, so the devs might've been doing some crazy stuff with the hardware.

**Comment by refractionpcsx2** [comment]
maybe :P Well if the positive VU rounding works for both, I'll put it in the gamedb so it's automatic.

**Comment by TheTechnician27** [comment]
So the positive VU rounding definitely works for Gauntlet Golf (I'm confused, but I'll accept your wizardry). I actually realized that I left something out in my description of how to recreate the second bug: like I said, the spaces where this happens are deterministic, and it's actually the first five spaces of the Tropical -> Long board where this can be seen most immediately. If you were playing on the Short or the Medium Tropical boards (for some reason, this doesn't affect Space or Medieval), the issue may take a while to show up because the spaces aren't necessarily right at the start; for the Long board, however, it definitely is.

**Comment by refractionpcsx2** [comment]
cool, well as long as it doesn't break anything else, that'll be a bonus :D

**Comment by TheTechnician27** [comment]
Yup! In the meantime, the walking animation will just remain as a funny little gag in casual play and a tiny hiccup for a speedrun. Were you able to recreate it when you specifically played on the Long board?

**Comment by refractionpcsx2** [comment]
Hmm, try setting the EE rounding to Positive.   I didn't replicate the skipping, but the disco dude kinda freaked out on the board until I set the EE rounding to Positive.  See if you can replicate it with that setting.

**Comment by TheTechnician27** [comment]
It worked. So positive VU rounding fixes Gauntlet Golf, and positive EE rounding fixes the jittery walking. I'll test positive EE rounding just by itself with VU rounding set to default just to see what happens, but those two definitely fix the issue.

**Comment by refractionpcsx2** [comment]
No, please leave them both on and play the game to make sure it's okay, if it is, I'll add it to the GameDB. Since they're both needed for different things, I'd like to know the game is stable with those settings :)

**Comment by TheTechnician27** [comment]
Sure thing. I just wanted to test it really quickly just in case EE actually resolved them both and VU was unnecessary. Here's something interesting: when leaving just positive EE rounding on but turning setting VU rounding to default, the drawbridges open and close *twice* instead of just once, but then the same issue presents itself.

**Comment by refractionpcsx2** [comment]
yeah I think I tried EE rounding first and it didn't change anything on the drawbridges, I expected it to be that over the VU's lol

**Comment by TheTechnician27** [comment]
Alright, I'll keep playing with both enabled like you said to make sure there are no stability issues – I'll play all three boards on Long (the Short and Medium boards are literally just exact subsets of the Long boards). The final verdict on these two issues are:

* Default rounding on both: no fix (duh).

* Default EE; positive VU: Gauntlet Golf fixed, jittering still there.

* Positive EE; default VU: Gauntlet Golf marginally but not really fixed, jittering fixed.

* Both positive: Gauntlet Golf fully fixed and jittering fully fixed.

These were the only two conspicuous issues I know of that this game had with PCSX2, so if anything new comes up, I'll report it back in this thread.

**Comment by refractionpcsx2** [comment]
cool, well I'll get one of the guys to add this to the game db :) Thanks for testing!

**Comment by TheTechnician27** [comment]
Thanks even more for finding the fix! I'll see if I can add a guide linking to the PCSX2 Wiki page detailing the fix (once it's live on the game db) on the game's speedrun.com page to make speedrunning the game more accessible.

**Comment by refractionpcsx2** [comment]
Now in Master, should be in dev build 1594

**Comment by TheTechnician27** [comment]
@refractionpcsx2 So the only issue with enabling positive rounding that I can see after a few hours of testing is what you described previously: characters on the start square and the staging squares (the giant ones with four spaces on them) sometimes jitter rapidly back and forth. However, this is pretty minor. I can test whether it's the EE or the VU rounding doing this (or if it's specifically a combination of both), but this graphical glitch is fairly minor, does not impair gameplay in any way, and is far preferable to the two glitches noted above.

Also, I speedrun this niche fever dream of a party game, so feel free to @ me if you ever need help testing it for some bizarre reason.

**Comment by refractionpcsx2** [comment]
Well, that's the lesser evil and I'm not sure we can make the game completely happy, maybe with a patch in future, but honestly I don't know where to start.

**Comment by TheTechnician27** [comment]
@refractionpcsx2 Yeah, it's basically nothing. The only reason I even bring it up is because you asked me to test for stability, but this is the only issue I could find, which is why [I think it's good enough to warrant a 'Perfect' score on the Wiki.](https://wiki.pcsx2.net/Pac-Man_Fever) Given this seems to have to do with PS2 float rounding which can probably only be "fixed" with a game-specific patch (I wouldn't even begin to know where to look in the debugger) or well into the future with insane PC hardware, this feels so, so low on the totem pole (extremely minor issue that does not impact gameplay whatsoever in an obscure game that otherwise plays perfectly) that even something like #3120 (*wink, wink*, *plug, plug*) is more worth your time.

TL;DR: Much, much lesser evil.

**Comment by refractionpcsx2** [comment]
Well thank you for testing.  The game is not worth a "perfect" score though, for a perfect score there would need to be 0 bugs, graphically, physically, anything.  At best the game is "Playable".

yeah maybe if it can be patched, then sure, but right now, it's gonna have to stay as Playable.

**Comment by TheTechnician27** [comment]
Alright, changed it to Playable. Anyway, I'll close this out by saying that if you ever need a memcard with all the mini-games unlocked for some reason, [there's now one up on the speedrun.com page](https://www.speedrun.com/pmf/resources). Thanks again for the help. =)

**Comment by refractionpcsx2** [comment]
Thanks for the report and testing :)
